### DISCUSSION TITLE
Implement menus like BotFather

### DISCUSSION THREAD

**Discussion by ghost** [nutgram/nutgram#337] [Q&A] [answered]
Hi 

How can i implement menus like BotFather's menus in Nutgram?

My bot has too many menus and buttons

Earlier, I had defined a `Command` and a separate class for each button, the number of classes had increased to more than 200, and it had made the development work much more difficult for me.

Please watch this video:

https://user-images.githubusercontent.com/78247242/220736997-cc765975-6f40-4e49-82d8-d75d06f69281.mp4

Recently I am trying to rewrite my codes and this time I am trying to use the `InlineMenu` feature in `Nutgram`, but it is a bit confusing and difficult for me to use. For example, getting values from the user and other things...

I tried to use [this file](https://github.com/Lukasss93/telegram-stickeroptimizer/blob/master/app/Telegram/Conversations/SettingsConversation.php) as a template for my work and proceed according to it, but it is not very interesting... ( for example the `reopen` property in the class )

In any case, the `InlineMenu` feature in `Nutgram` does not seem to be the exact solution of what I want (that is, it seems that it is not possible to create menus exactly similar to BotFather's menus and its functionality; since it is a type of `Conversation`, it seems that only the last open menu can be used by the user )

Anyway, How can i implement menus like BotFather's menu exactly with Nutgram ?

Any menu sent to the user at any time can be used, in BotFather

Can you send a sample code for this ?

**Comment by ghost** [comment]
????????????????????????????? ☹️

**Comment by Lukasss93** [comment, marked as answer]
I don't think this feature will be added in the near future but you can create it in your bot :)

**Reply by ghost** [reply]
I can implement it with Commands. But this way the number of classes increases a lot.
Do you know a better way?
Is it possible to implement this with InlineMenu?

**Reply by ghost** [reply]
Oh damn. I can put all commands in one class...! Instead of creating a class for each :/
```php
$bot->onCommand('account', AccountCommand::class);
$bot->onCallbackQueryData('account/transactions', [AccountCommand::class, 'transactions']);
$bot->onCallbackQueryData('account/wallet', [AccountCommand::class, 'wallet']);
// ....
```
